lib/signet/oauth_2/client.rb

```diff
@@ -823,6 +828,25 @@ def expires_at= new_expires_at
         @expires_at = normalize_timestamp new_expires_at
       end
 
+      ##
+      # Returns the scopes granted by the authorization server.
+      #
+      # @return [Array] The scope of access returned by the authorization server.
+      def granted_scopes
+        @granted_scopes
+      end
+
+      ##
+      # Sets the scopes returned by authorization server for this client.
+      #
+      # @param [Array] new_granted_scopes
+      #   The scope of access the client is requesting.  This may be
+      #   expressed as either an Array of String objects or as a
+      #   space-delimited String.
+      def granted_scopes= new_granted_scopes
+        @granted_scopes = new_granted_scopes&.split
```

Comment:
If an Array is passed in for `new_granted_scopes` (see your docs above), it won't respond to the message `split`.

Reply:
Yes. I have updated the docs to reflect the expected behavior here.

Reply:
I'd suggest allowing Array to be passed. Otherwise, `granted_scopes=` takes a String but `granted_scopes` returns an Array, which would be confusing and invite buggy usage.

Reply:
I didn't want this method to accept `Array` since we get space delimited `String` in response and this method should ideally not be used to update `granted_scopes` manually.
But for the sake of not introducing bugs, I'll make the change.